libraries/lib-audio-devices/DeviceChange.cpp

```diff
@@ -348,8 +348,9 @@ class DeviceChangeListener final : public DeviceChangeInterface
          {
             This->mEnabled = false;
             wxMutexGuiEnter();
```

Comment:
Do we still need this mutex? Looks like we now run the code strictly on the UI thread

Reply:
> Do we still need this mutex? Looks like we now run the code strictly on the UI thread

Firstly, EXPERIMENTAL_DEVICE_CHANGE_HANDLER is off by default.  I was being complete, trying to maintain compilability in case it is on, but I don't know if we will turn it on.

Secondly, I realize now that I only kept macOS compilable and had to do more.

Thirdly, no, the calls to publish the event happen on different threads, obviously with GTK where pthreads is used to spawn own.

Fourthly though, yes, even on other threads, wxMutexGuiEnter is not necessary to protect calls to wxEvtHandler::CallAfter (what is wrapped inside BasicUI::CallAfter), according to the documentation of CallAfter itself, as safe to call from any thread.

Reply:
So, I will remove the calls, as you suggested, and also fix the build of other platforms in case this EXPERIMENTAL is on.